orcestra/postprocess/level1.py

```diff
@@ -0,0 +1,187 @@
+import numpy as np
+import xarray as xr
+import json
+import os
+
+# Get the directory of the current script
+script_dir = os.path.dirname(os.path.abspath(__file__))
+
+# Construct the path to config.json relative to the script's location
+config_path = os.path.join(script_dir, "config.json")
+
+# read config.json file for parameters
+with open(config_path) as f:
+    config = json.load(f)
+
+
+def _noise_filter_radar(ds):
+    """Filter radar data by noise level.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Level0 radar dataset.
+
+    Returns
+    -------
+    xr.Dataset
+        Radar data filtered by noise level.
+
+    """
+    return ds.where(ds.npw1 > config["noise_threshold"])
+
+
+def _state_filter_radar(ds):
+    """Filter radar data for valid state.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Level0 radar dataset.
+
+    Returns
+    -------
+    xr.Dataset
+        Radar data filtered for valid state.
+    """
+
+    return ds.where(ds.grst == config["valid_radar_state"])
+
+
+def _roll_filter(ds, ds_bahamas):
+    """Filter any dataset by plane roll angle.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Level0 dataset.
+    ds_bahamas : xr.Dataset
+        Level0 BAHAMAS dataset.
+
+    Returns
+    -------
+    xr.Dataset
+        Dataset filtered by plane roll angle.
+    """
+
+    ds_bahamas_subsampled = ds_bahamas.sel(
+        time=ds.time, method="nearest"
+    ).assign_coords(time=ds.time)
+
+    return ds.where(np.abs(ds_bahamas_subsampled.IRS_PHI) < config["roll_threshold"])
+
+
+def _altitude_filter(ds, ds_bahamas):
+    """Filter any dataset by plane altitude.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Level0 dataset.
+    ds_bahamas : xr.Dataset
+        Level0 BAHAMAS dataset.
+
+    Returns
+    -------
+    xr.Dataset
+        Dataset filtered by plane altitude.
+    """
+
+    ds_bahamas_subsampled = ds_bahamas.sel(
+        time=ds.time, method="nearest"
+    ).assign_coords(time=ds.time)
+
+    return ds.where(ds_bahamas_subsampled.IRS_ALT > config["altitude_threshold"])
+
+
+def coarsen_radiometer(ds):
+    """Coarsen radiometer data to 1 Hz.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Level0 radiometer dataset.
+
+    Returns
+    -------
+    xr.Dataset
+        Radiometer data coarsened to 1 Hz.
```

Comment:
I talked to @d70-t about this and the measurements seem to be real data and we don't want to average them right away. The fact that the time axis is broken is known, but the vendor won't fix it. Maybe the best way would be to construct a corrected time coordinate... but I think we need to do some more questionnaires with the scientists to understand this better.

Reply:
Ok, sounds good! I checked in Henning's code, and they resample the radiometer data to 1Hz, but this doesn't mean it's correct. I decided to go with coarsening instead of resample or groupby since it is much faster. However, I realised I had to update this slightly since in some cases there are still duplicates left in time (see latest merge request).